File: 백준/Silver/11047. 동전 0/동전 0.swift
let input = readLine()!.split(separator: " ").map { Int($0)! }
let n = input[0]
var k = input[1]

var coins: [Int] = []
var result: Int = 0

for _ in 0..<n {
    coins.append(Int(readLine()!)!)
}

for coin in coins.reversed() {
    if k >= coin {
        let a = k / coin
        k = k % coin
        result += a
    }
}

print(result)

File: 백준/Silver/11047. 동전 0/동전 0.cc
#include <stdio.h>

int main() {
    int N, K;
    scanf("%d %d", &N, &K);

    int coin[10];
    for (int i = 0; i < N; i++) {
        scanf("%d", &coin[i]);
    }

    int result = 0;

    for (int i = N - 1; i >= 0; i--) {
        if (K >= coin[i]) {
            result += K / coin[i];
            K %= coin[i];
        }
    }

    printf("%d\n", result);
    return 0;
}